apps/annotations/views/tag_views.py

```diff
@@ -61,17 +61,26 @@ class EditTag(UpdateView, PermissionRequiredMixin):
     }
 
     def get_queryset(self):
-        return Tag.objects.filter(team=self.request.team)
+        return Tag.objects.filter(team=self.request.team, is_system_tag=False)
 
     def get_success_url(self):
         return reverse("annotations:tag_home", args=[self.request.team.slug])
 
+    def get(self, request, *args, **kwargs):
+        pk = kwargs.get("pk")
+        queryset = self.get_queryset()
+        if not queryset.filter(pk=pk).exists():
+            return HttpResponseRedirect(reverse("annotations:tag_home", args=[self.request.team.slug]))
```

Comment:
FYI: Django has a shortcut for doing redirects: https://docs.djangoproject.com/en/5.0/topics/http/shortcuts/#redirect

```suggestion
            return redirect("annotations:tag_home", team_slug=self.request.team.slug)
```

Reply:
[85486cd](https://github.com/dimagi/open-chat-studio/pull/440/commits/85486cdf2fd21a52de2afcf8bb20f0b6b2fce8da)